### PR TITLE
Use ResponseHelper.Error() for all error responses

### PR DESCRIPTION
**Pull request by lmars**

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -60,14 +59,12 @@ func (r *AppRepo) Add(data interface{}) error {
 	return err
 }
 
-var ErrNotFound = errors.New("controller: resource not found")
-
```
Comment (titanous) [inline app.go:64]: Why did you change `ErrNotFound` into a type?

Comment (lmars) [inline app.go:64]: Mainly so I could slot it into [this switch statement](https://github.com/flynn/flynn-controller/blob/improve-error-handling/controller.go#L93), but I guess I could just put an `if err == ErrNotFound` in the `default` clause?

Comment (lmars) [inline app.go:64]: Yes it seems wrong to make it a type, I have reverted that change: 33067e0

**Comment by titanous** [comment]
This is awesome, thanks!
